src/main/java/bookrepo/controller/CategoryController.java

```diff
@@ -0,0 +1,84 @@
+package bookrepo.controller;
+
+import bookrepo.dto.book.BookDtoWithoutCategoryIds;
+import bookrepo.dto.category.CategoryDto;
+import bookrepo.exception.EntityNotFoundException;
+import bookrepo.mapper.BookMapper;
+import bookrepo.model.Book;
+import bookrepo.model.Category;
+import bookrepo.repository.book.BookRepository;
+import bookrepo.repository.category.CategoryRepository;
+import bookrepo.service.CategoryService;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/categories")
+@RequiredArgsConstructor
+@Tag(name = "Category Controller", description = "Endpoints for managing categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+    private final CategoryRepository categoryRepository;
+
+    @PreAuthorize("hasAuthority('ADMIN')")
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public CategoryDto createCategory(@RequestBody @Valid CategoryDto categoryDto) {
+        return categoryService.save(categoryDto);
+    }
+
+    @PreAuthorize("hasAuthority('USER')")
+    @GetMapping
+    public List<CategoryDto> getAll() {
```

Comment:
use Pageable pageable
```suggestion
    public Page<CategoryDto> getAll() {
```